spec-tests/src/test/java/org/candlepin/spec/activationkey/ActivationKeySpecTest.java

```diff
@@ -0,0 +1,651 @@
+            .ignoringFields("created")
+            .ignoringFields("updated")
+            .ignoringFields("name")
+            .isEqualTo(keyFiltered);
+        assertThat(keyFiltered)
+            .hasFieldOrPropertyWithValue("name", null);
+    }
+
+    @Test
+    public void shouldSetAutoAttachOnActivationKey() {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = createOwner(adminClient);
+
+        ActivationKeyDTO activationKey = createActivationKey(
+            adminClient, owner, ActivationKeys.random(owner).autoAttach(null));
+        assertThat(activationKey)
+            .hasFieldOrPropertyWithValue("autoAttach", null);
+
+        ActivationKeyDTO keyWithAutoAttach = createActivationKey(
+            adminClient, owner, ActivationKeys.random(owner).autoAttach(true));
+        assertThat(keyWithAutoAttach)
+            .hasFieldOrPropertyWithValue("autoAttach", true);
+
+        ActivationKeyDTO keyWithoutAutoAttach = createActivationKey(
+            adminClient, owner, ActivationKeys.random(owner).autoAttach(false));
+        assertThat(keyWithoutAutoAttach)
+            .hasFieldOrPropertyWithValue("autoAttach", false);
+    }
+
+    @Test
+    public void shouldAllowOwnersToListExistingActivationKeys() {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = createOwner(adminClient);
+
+        createActivationKey(adminClient, owner);
+        createActivationKey(adminClient, owner);
+        createActivationKey(adminClient, owner);
+
+        List<ActivationKeyDTO> listOfActivationKeys =
+            adminClient.owners().ownerActivationKeys(owner.getKey(), null);
+        assertThat(listOfActivationKeys)
+            .hasSize(3);
+    }
+
+    @Test
+    public void shouldAllowUpdatingOfNames() {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = createOwner(adminClient);
+        OwnerDTO anotherOwner = createOwner(adminClient);
+        ApiClient userClient = createUserClient(adminClient, owner);
+        ApiClient anotherUserClient = createUserClient(adminClient, anotherOwner);
+        ActivationKeyDTO activationKey = createActivationKey(adminClient, owner);
+
+        String randomName1 = StringUtil.random("name1");
+        activationKey.name(randomName1);
+        ActivationKeyDTO activationKeyUpdated = updateActivationKey(adminClient, activationKey);
+        assertThat(activationKeyUpdated)
+            .hasFieldOrPropertyWithValue("name", randomName1);
+
+        String randomName2 = StringUtil.random("name_2");
+        activationKey.name(randomName2);
+        activationKeyUpdated = updateActivationKey(userClient, activationKey);
+        assertThat(activationKeyUpdated)
+            .hasFieldOrPropertyWithValue("name", randomName2);
+
+        String randomName3 = StringUtil.random("not-gonna-happen");
+        activationKey.name(randomName3);
+        assertNotFound(() -> updateActivationKey(anotherUserClient, activationKey));
```

Comment:
This seems like it could almost be a separate test. Validating that a client of a different owner cannot update activation keys that belong to a different owner. This can also apply to some of the other tests.

Reply:
done